src/ray_serializer.jl

```diff
@@ -84,13 +84,23 @@ end
 
 function deserialize_from_ray_object(ray_obj::SharedPtr{ray_jll.RayObject},
                                      outer_object_ref=nothing)
+    data = ray_jll.get_data(ray_obj)
     metadata = ray_jll.get_metadata(ray_obj)
+
+    # Metadata indicates an error was reported from the raylet.
```

Comment:
(I'm okay with punting on this for now, but we should leave a breadcrumb here
```suggestion
    # If the raylet reports an error, metadata will be set with the error code.
    # TODO: handle non-error metadata as well
```

Reply:
Makes sense. I was aware that having metadata doesn't always indicate an error but the current implementation treats it as such so that's how I documented it. I'll bring things more up to speed with reality